packages/client/tests/e2e/_utils/run.ts

```diff
@@ -84,20 +91,63 @@ async function main() {
 
   console.log('🐳 Starting tests in docker')
   // tarball was created, ready to send it to docker and begin e2e tests
-  const testNames = args._.join(' ').replace(/\//g, '-')
-  await $`docker compose -f ${__dirname}/docker-compose.yml down --remove-orphans`
-  await $`docker compose -f ${__dirname}/docker-compose.yml build ${testNames}`
-  await $`docker compose -f ${__dirname}/docker-compose.yml up ${testNames}`
+  const testStepFiles = await glob('../**/_steps.ts', { cwd: __dirname })
+  let e2eTestNames = testStepFiles.map((p) => path.relative('..', path.dirname(p)))
+
+  if (args._.length > 0) {
+    e2eTestNames = e2eTestNames.filter((p) => args._.some((a) => p.includes(a)))
+  }
+
+  const dockerVolumes = [
+    `${path.resolve(__dirname, '..')}:/e2e`,
+    `${path.resolve(__dirname, '..', '..', '..')}:/client`,
+    `${path.resolve(__dirname, '..', '..', '..', '..', '..')}:/repo`,
+    `${path.resolve(__dirname, '..', '.cache')}:/root/.cache`,
+    `${(await $`pnpm store path`.quiet()).stdout.trim()}:/root/.local/share/pnpm/store/v3`,
+  ]
+  const dockerVolumeArgs = dockerVolumes.map((v) => `-v ${v}`).join(' ')
+
+  await $`docker build -f ${__dirname}/standard.dockerfile -t prisma-e2e-test-runner .`
+
+  const dockerJobs = e2eTestNames.map((path) => {
+    return async () =>
+      await $`docker run --rm ${dockerVolumeArgs.split(' ')} -e "NAME=${path}" prisma-e2e-test-runner`.nothrow()
+  })
+
+  let jobResults: (ProcessOutput & { name: string })[] = []
+  if (args['--runInBand'] === true) {
+    console.log('🏃 Running tests in band')
+    for (const [i, job] of dockerJobs.entries()) {
+      console.log(`💡 Running test ${i + 1}/${dockerJobs.length}`)
+      jobResults.push(Object.assign(await job(), { name: e2eTestNames[i] }))
+    }
+  } else {
+    console.log('🏃 Running tests in parallel')
+    jobResults = (await Promise.all(dockerJobs.map((job) => job()))).map((result, i) => {
+      return Object.assign(result, { name: e2eTestNames[i] })
+    })
+  }
+
+  const failedJobResults = jobResults.filter((r) => r.exitCode !== 0)
+  const passedJobResults = jobResults.filter((r) => r.exitCode === 0)
+
+  if (args['--verbose'] === true) {
+    for (const result of failedJobResults) {
+      console.log(`🛑 ${result.name} failed with exit code`, result.exitCode)
+      await $`cat ${path.resolve(__dirname, '..', result.name, 'LOGS.txt')}`
+      await sleep(50)
```

Comment:
Could you add a comment here about why this sleep is necessary?

Reply:
It was a strange behavior on CI, that I do not fully understand which would prevent proper log output. I think it has something to do with zx, but didn't get the time to debug it. I'll leave a comment.